### PR TITLE
Preserve bold text when pasted

### PR DESCRIPTION
**Pull request by ivan-i**
Fixes https://github.com/quilljs/quill/issues/306

### CODE REVIEW

File: src/core/normalizer.coffee [hunk 1]
```diff
@@ -57,9 +57,10 @@ class Normalizer
         node.removeAttribute(attribute)
     )
     # Chrome turns <b> into style in some cases
-    if (node.style.fontWeight == 'bold')
+    if (node.style.fontWeight == 'bold' or node.style.fontWeight > 500)
```
Comment (jhchen) [inline src/core/normalizer.coffee]: Can you change this to > 400? Technically a font-weight of 500 is also considered bolded.

**Comment by ivan-i** [comment]
After some reading I am not sure that > 400 is correct. From http://destination-code.blogspot.com/2009/01/font-weight-number-keywords-100-900.html:

> The keyword 400 is equivalent to the font weight regular or normal. The keyword 700 is equivalent to the font weight bold. For a particular typeface, if the other keywords are not assigned, keywords **100 to 500 are equivalent to the font weight regular or normal** and keywords 600 to 900 are equivalent to the font weight bold.

Also:

> When the font is a True Type font, Firefox 3 renders the number keywords 100, 200, 300, 400 and 500 with the regular or normal font and the keywords 600, 700, 800 and 900 with the bold font

> Safari 3.2 renders text of weights 100 to 500 in regular fonts and text of font weight 600 to 900 in boldface fonts, regardless of whether the appropriate font weights are available in the typeface or not

Therefore I think `> 500` was actually correct.

**Comment by jhchen** [comment]
You are correct thank you for looking into it and the PR.

**Comment by aadamsx** [comment]
I'm running into this issue now.  I have text in word, and some of it is bolded, when I copy and paste it the bolded text reverts to normal.

**Comment by aadamsx** [comment]
Here's a little more background on what I'm seeing:

I'm using Quill Editor to provide rich text editing into my application.

In mongo when I save off a simple edit session, it looks something like this:

```
<div class="ql-editor ql-content" contenteditable="true">This</div><div class="ql-editor ql-content" contenteditable="true"><br></div><div class="ql-editor ql-content" contenteditable="true">text</div><div class="ql-editor ql-content" contenteditable="true"><br></div><div class="ql-editor ql-content" contenteditable="true">is</div><div class="ql-editor ql-content" contenteditable="true"><br></div><div class="ql-editor ql-content" contenteditable="true"><b>BOLD</b></div><div class="ql-paste-manager" contenteditable="true"><br></div>
```

Using Meteor Blaze and Handlebars my code to target this html looks like this:

```
                      <div class="contents regular">
                        <div id="message-{{messageId}}">
                          {{{message}}}
                        </div>
                      </div>
                      <span id="copy-{{messageId}}" alt="Copy to clipboard" data-clipboard-target="#message-{{messageId}}" data-id="{{messageId}}" class="" style="float: right; margin-right: 15px; cursor: pointer;">
                        <i class="fa fa-clipboard"></i>
                      </span>
```
I also have code to allow me to see what is copied to the clipboard:
```
  var clipboard = new Clipboard('span[id^="copy-"]');
  clipboard.on('success', function (e) {
    console.info('Action:', e.action);
    console.info('Text:', e.text);
    console.info('Trigger:', e.trigger);
  });
```

When I copy the text to the clipboard, all I see in this example is this in the console:
```
Action: copy
messages-detail.js?005609202faf1271a61336d16a44247de0aa130c:30 Text: 
This

text

is

BOLD
```

I don't/can't see the "behind-the-seens" formatting going on in the Text attribute.  So when I copy this text from the clipboard to Word, the word BOLD is bolded, but if I copy this from the clipboard back into Quill editor, all the formatting and any color stays, but I loose the bold in the BOLD text.

Any help see what formatting is going on in the clipboard is appreciated.

**Comment by aadamsx** [comment]
After further research. The bold seems to be working in FF but not in Chrome.  

I have another example...

When I copy from Chrome rich-text (with embedded HTML) to the Quill Editor, looking at the binary of the clipboard, the bolded text has this for embedded HTML:
```
<b style="box-sizing: border-box; -webkit-font-smoothing: antialiased; font-weight: 700; color: rgb(204, 0,0); font-family: Arial, sans-serif; font-size: 12pt; background-color: inherit;"><br class="Apple-interchange-newline">*** Bold Text ***Â</b>
```

In FF the binary for the text looks like this:
```
<b style="color: rgb(204, 0,0); font-family: Arial, sans-serif; font-size: 12pt;">*** Bold Text ***</b>
```
In Chrome the *** Bold Text *** is not bolded, in FF it is.

This is the mongo field saved from Quill when in Chrome:
```
<div class="ql-editor ql-content" contenteditable="true"><br></div><div class="ql-editor ql-content" contenteditable="true"><span style="color: rgb(204, 0, 0); font-family: Arial, sans-serif; font-size: 12pt;">*****Please make sure to read this entire message before proceeding*****&nbsp;</span></div><div class="ql-paste-manager" contenteditable="true"><br></div>
```

This is the mongo field saved from Quill when in FF:
```
<div class="ql-editor ql-content" contenteditable="true"><b style="color: rgb(204, 0, 0); font-family: Arial,sans-serif; font-size: 12pt;">*****Bold Text*****&nbsp;</b></div><div class="ql-editor ql-content" contenteditable="true"><br></div><div class="ql-paste-manager" contenteditable="true"><br></div>
```

**Comment by jhchen** [comment]
First this PR addresses an issue where Quill loses bolding pasted within itself. Word is out of scope of this PR. That being said it still seems to work. If I copy bolded text or a sentence with bolded words from Word 2016, the formatting stays when pasted into Chrome on the quilljs.com editor. I'm not clear on what interaction there is between clipboard.js and Quill but if you can reduce and reproduce the issue outside of Meteor and clipboard.js, you should make a separate Github issue with reproduction steps.

**Pull request merged**